webapplication/user/views.py

```diff
@@ -22,3 +26,16 @@ def get(self, *args, **kwargs):
         except Http404:
             raise NotFound(_("Email verification failed"))
         return Response(data=_("Email has been successfully confirmed!"), status=HTTP_200_OK)
+
+
+class TransactionListAPIView(ListAPIView):
+    permission_classes = (IsAuthenticated, )
+    queryset = Transaction.objects.all()
+    serializer_class = TransactionSerializer
+    pagination_class = None
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        if self.request.user.is_staff:
+            return queryset
+        return queryset.filter(user=self.request.user)
```

Comment:
Should be the same approach as here https://github.com/QuantuMobileSoftware/geoap/commit/94b741782f4d2128aa275c40e8e4ba50b85cb9e7